tests/Tests/400_ods_dashboard/400_ods_dashboard.robot

```diff
@@ -1,45 +1,49 @@
-    ${len}=  Get Length    ${link_elements}
+    Sleep    5
+    ${link_elements}=    Get WebElements    //a[@class="odh-card__footer__link" and not(starts-with(@href, '#'))]
+    ${len}=    Get Length    ${link_elements}
     Log To Console    ${len} Links found\n
-    FOR  ${idx}  ${ext_link}  IN ENUMERATE  @{link_elements}  start=1
-        ${href}=  Get Element Attribute    ${ext_link}    href
-        ${status}=  Check HTTP Status Code   link_to_check=${href}
+    FOR    ${idx}    ${ext_link}    IN ENUMERATE    @{link_elements}    start=1
+        ${href}=    Get Element Attribute    ${ext_link}    href
+        ${status}=    Check HTTP Status Code    link_to_check=${href}
         Log To Console    ${idx}. ${href} gets status code ${status}
     END
 
 Verify Content In RHODS Explore Section
-    [Documentation]  It verifies if the content present in Explore section of RHODS corresponds to expected one.
-    ...              It compares the actual data with the one registered in a JSON file. The checks are about:
-    ...              - Card's details (text, badges, images)
-    ...              - Sidebar (titles, links text, links status)
-    [Tags]  Sanity
-    ...     ODS-488  ODS-993
-    ${EXP_DATA_DICT}=   Load Expected Data Of RHODS Explore Section
+    [Documentation]    It verifies if the content present in Explore section of RHODS corresponds to expected one.
+    ...    It compares the actual data with the one registered in a JSON file. The checks are about:
+    ...    - Card's details (text, badges, images)
+    ...    - Sidebar (titles, links text, links status)
+    [Tags]    Sanity    ODS-488    ODS-993
+    ${EXP_DATA_DICT}=    Load Expected Data Of RHODS Explore Section
     Click Link    Explore
     Wait Until Cards Are Loaded
-    Check Number Of Displayed Cards Is Correct  expected_data=${EXP_DATA_DICT}
-    Check Cards Details Are Correct   expected_data=${EXP_DATA_DICT}
+    Check Number Of Displayed Cards Is Correct    expected_data=${EXP_DATA_DICT}
+    Check Cards Details Are Correct    expected_data=${EXP_DATA_DICT}
+
+Verify logged in users are displayed in the dashboard
+    [Documentation]    It verifies that logged in users username is displayed on RHODS Dashboard.
+    [Tags]    Sanity    ODS-354
+    Element Text Should Be    xpath=//div[@class='pf-c-page__header-tools-item'][3]//span[1]    ${TEST_USER.USERNAME}
 
 
 *** Keywords ***
 Dashboard Test Setup
-  Set Library Search Order  SeleniumLibrary
-  Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
-  Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
-  Wait for RHODS Dashboard to Load
+    Set Library Search Order    SeleniumLibrary
+    Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
```

Comment:
I think you can replace this with `Begin Web Test`

Reply:
But Begin Web Test is opening JupyterHub which is not required here.

Reply:
`Beign Web Test` goes to JupyterHub to verify that the spawner page is in a good state a no server is currently running, but its last step is to `Go To  ${ODH_DASHBOARD_URL}`, getting you to where you need to be to `Wait for RHODS Dashboard to Load`

Reply:
I think it's okay to use `Dashboard Test Setup`. it is the one used by all the TCs in this component.

Reply:
>it is the one used by all the TCs in this component

Then it should be abstracted as a common keyword rather than be redefined in each TC

Reply:
it is a common keyword and it is set as Test Setup. Of course it should be put in the resource file as part of refactoring process

Reply:
I don't think it should be here in the TC at all.
If the keyword is already defined as a common keyword in a resource file then that resource file should be imported and that version used.
If it is not already defined in a resource file, then it should be and this should be removed. Merging new code that needs to be refactored doesn't make sense IMHO.

Reply:
>> If it is not already defined in a resource file, then it should be and this should be removed. Merging new code that needs to be refactored doesn't make sense IMHO.


agree with that. The resource file has already the `Launch Dashboard` which would replace line 44-45-46. So @prudhvi-165 when moving this keyword to the Resource file, pls integrate it with the `Launch Dashboard` keyword.
Works for you @lugi0 ?